src/System.Diagnostics.Tracing.Logger/src/System/Diagnostics/Tracing/Logger.cs

```diff
@@ -0,0 +1,206 @@
+    ///      in that appdomain/process.   Thus there can only be one EventSource call 'MyProgram'.   This is a
+    ///      problem if like ASP.NET, which wishes to support many independent tenants in the same appdomain.  
+    ///      Because the events form one tenant will mix with the others.  
+    ///   2) Dependency Injection - Logger supports ILogger, which allows you to build your application so that
+    ///      the exact implemenation of the logging system can be passed into it.   Logger has support for this.
+    /// 
+    /// If you don't have these requirements you should seriously consider using EventSource instead.   However
+    /// if you are using ASP.NET 5, they support both (1) and (2) above, and it is best to stick with Logger.  
+    /// 
+    /// Logger leverages DiagnosticListener as its 'plumbing'.   Thus Every Logger is a DiagnosticListener which  
+    /// means that you can discover them using DiagnosticListener.AllListeners.   Once you have the Logger you want, 
+    /// you can use the Subscribe() method to connect to the stream of data.  
+    /// 
+    /// System.Diagnostics.Tracing.EventSource will have plumbing to treat all DiagnosticSource (and thus all 
+    /// Loggers), as EventSources (as long as the payloads are serializable).   Thus it is also possible access
+    /// the data stream of Loggers from an EventListener and therefore ETW and LTTng.  
+    /// </summary>
+    public class Logger : DiagnosticListener, ILogger
+    {
+        /// <summary>
+        /// Creates a new logger with the given name.   A Logger is a DiagnosticSource which understands LogLevel (verbosity levels)
+        /// </summary>
+        /// <param name="name">The name of the logger.  This is often the comonent name.</param>
+        /// <param name="maxLevel">The most verbose message that will ever be logged.  It defaults to Debug which, being the most
+        /// verbose level, means any message will be logged.   Useful in the context of security to make it less likely that 
+        /// private (debug) information does not get logged.  It also increases the efficienctly of IsEnabled if you only wish
+        /// to EVER turn low-verbosity events.</param>
+        public Logger(string name, LogLevel maxLevel = LogLevel.Debug) : base(name)
+        {
+            _maxLevel = maxLevel;  
+        }
+
+        /// <summary>
+        /// Returns true if 'level' is less verbose (more important) than the current level for the Logger.  This
+        /// should be used on all hot code paths before 'Log' is called so that even argument setup is skipped if
+        /// logging is off.   
+        /// </summary>
+        public virtual bool IsEnabled(LogLevel level)
+        {
+            if (level > _maxLevel)
+                return false;
+            if (_filter != null)
+            {
+                foreach (Func<string, LogLevel, bool> filter in _filter.GetInvocationList())
```

Comment:
Every invocation of _filter.GetInvocationList() is going to allocate a new array, and since IsEnabled is meant to be cheap, that seems relatively expensive.  Instead of having _filter be ```Func<string, LogLevel, bool>```, how about making it a ```List<Func<string, LogLevel, bool>>```?